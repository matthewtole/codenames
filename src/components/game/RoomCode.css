.game__room-code.modal .modal-content {
  width: 100%;
}

.game__room-code .hero {
  background-color: #2196f3;
}
