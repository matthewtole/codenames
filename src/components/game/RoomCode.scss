@import '../../sass/_variables';

.game__room-code.modal .modal-content {
  width: 100%;
}

.game__room-code .hero {
  background-color: $blue-spy--dark;
}
