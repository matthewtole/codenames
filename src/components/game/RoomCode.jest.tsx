import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { RoomCode, RoomCodeProps } from './RoomCode';
import { shallow } from 'enzyme';

describe('RoomCode', () => {
  const onClose = jest.fn();
  const defaultProps: RoomCodeProps = {
    code: '123456',
    onClose,
  };

  beforeEach(() => {
    onClose.mockReset();
  });

  it('should render correctly', () => {
    const tree = renderer.create(<RoomCode {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call onClose when it gets clicked', () => {
    const roomCode = shallow(<RoomCode {...defaultProps} />);
    expect(onClose).not.toHaveBeenCalled();
    roomCode.simulate('click');
    expect(onClose).toHaveBeenCalled();
  });
});
