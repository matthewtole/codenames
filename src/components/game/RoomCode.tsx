import * as React from 'react';
import './RoomCode.scss';

interface Props {
  code: string;
  onClose: () => void;
}

interface State {}

export class ModalRoomCode extends React.Component<Props, State> {
  render() {
    const { code } = this.props;
    return (
      <div
        className="game__room-code modal is-active"
        onClick={() => this.props.onClose()}
      >
        <div className="modal-background" />
        <div className="modal-content">
          <section className="hero">
            <div className="hero-body">
              <div className="container has-text-centered">
                <h1 className="title has-text-white">{code}</h1>
              </div>
            </div>
          </section>
        </div>
      </div>
    );
  }
}
