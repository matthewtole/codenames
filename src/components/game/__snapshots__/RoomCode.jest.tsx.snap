// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomCode should render correctly 1`] = `
<div
  className="game__room-code modal is-active"
  onClick={[Function]}
>
  <div
    className="modal-background"
  />
  <div
    className="modal-content"
  >
    <section
      className="hero"
    >
      <div
        className="hero-body"
      >
        <div
          className="container has-text-centered"
        >
          <h1
            className="title has-text-white"
          >
            123456
          </h1>
        </div>
      </div>
    </section>
  </div>
</div>
`;
